### PR TITLE
FEATURE: Dynamic facilities page 🔨

### PR DESCRIPTION
**Pull request by Simperfy**
🐕 Dynamic facilities can be filtered using the search bar.
🐕 Did some refactoring (moved _buildMeter class and alike to a dedicated file).
🐩 All search suggestion is hard coded atm(can be found in utils folder).

### CODE REVIEW

File: lib/ui/pages/views/facilities_view/facilities_view.dart [hunk 1]
```diff
@@ -88,205 +83,75 @@ class _HospitalHeaderCard extends ViewModelWidget<FacilitiesViewModel> {
...
+                      SizedBox(height: 5),
+                      // TODO: This can be improved using a loop
+                      FacilitiesTitle(
+                          title: 'Facilities(covid):', enableLegends: true),
+                      FacilitiesDetailsRow(
+                        occupancyRate: model.calculateOccupancyRate(model.hospitalSummary?.icuVacant, model.hospitalSummary?.icuOccupied),
+                        title: 'icu beds:',
+                        occupied: model.hospitalSummary?.icuOccupied,
+                        vacant: model.hospitalSummary?.icuVacant,
+                      ),
```
Comment (DauntlessDev) [inline lib/ui/pages/views/facilities_view/facilities_view.dart:109]: In `calculateOccupancyRate` method, occupied is the parameter and the second one is vacant, that's why we need to interchange the parameters' position for the `FacilitiesDetailsRow` widget.

**Review by DauntlessDev — COMMENTED** [review]
Great, all good 👍 We just need to fix the color of the meter.

File: lib/ui/pages/views/facilities_view/facilities_view.dart (hunk 1 above)
Comment (Simperfy) [reply lib/ui/pages/views/facilities_view/facilities_view.dart:109]: Will definitely change to this.
```diff
-occupancyRate: model.calculateOccupancyRate(model.hospitalSummary?.icuVacant, model.hospitalSummary?.icuOccupied),
+occupancyRate: model.calculateOccupancyRate(model.hospitalSummary?.icuOccupied, model.hospitalSummary?.icuVacant),
```

**Review by DauntlessDev — APPROVED** [review]
Nice all concerns were fixed! 🥇

**Pull request merged**